src/NuGet.Core/NuGet.Protocol/PackagesFolder/LocalPackageFileCache.cs

```diff
@@ -95,6 +108,27 @@ public virtual bool Sha512Exists(string sha512Path)
             return exists;
         }
 
+        internal void UpdateLastAccessTime(string nupkgMetadataPath)
+        {
+            var exists = _metadataFileCache.ContainsKey(nupkgMetadataPath);
+            if (exists) return;
+
+            try
+            {
+                File.SetLastAccessTimeUtc(nupkgMetadataPath, DateTime.UtcNow);
+                _metadataFileCache.TryAdd(nupkgMetadataPath, true);
+            }
+            catch (Exception ex)
+            {
+                if (_logger != null)
+                {
+                    _logger.Log(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1802,
+                        string.Format(CultureInfo.CurrentCulture, /*NuGet.Commands.*/"Strings.Error_CouldNotUpdateMetadataLastAccessTime",
```

Comment:
You can just duplicate the string in this assembly.

Reply:
Is the LocalPackageFileCache used anywhere else? Want to make sure we don't get weirdness in other scenarios.

Reply:
It is referenced in
- `RestoreCommandProviders`, 
- `RestoreCommandProvidersCache`,
- `SourceRepositoryDependencyProvider`,
- `LocalV3FindPackageByIdResource`,
- `NuGetv3LocalRepository`.

What kind of weird issues do you have in mind?